onnxruntime/core/providers/cpu/tensor/upsample.cc

```diff
@@ -509,7 +517,19 @@ void ResizeBiCubic(
         }
 
         auto y_int = static_cast<int64_t>(std::floor(in_y));
-        auto& coeff_y = cubic_coeffs[in_y - y_int];
+        auto& coeff_y = exclude_outside ? y_coeff_holder : cubic_coeffs[in_y - y_int];
+        y_coeff_sum = 1;
+
+        if (exclude_outside) {
+          // When true, the weight of sampling locations outside the grid will be set to 0
+          // and the weight will be renormalized so that their sum is 1.0
+          y_coeff_sum = 0;
+          auto& orig_y_coeffs = cubic_coeffs[in_y - y_int];
+          for (int64_t i = 0, y_val = y_int - 1; y_val <= y_int + 2; y_val++, i++) {
+            y_coeff_holder[i] = (y_val < 0 || y_val >= static_cast<float>(input_height)) ? 0.0f : orig_y_coeffs[i];
```

Comment:
Spec isn't clear on what counts as outside. Should it be for a centered box?

Reply:
I agree spec is not clear for this! also not sure I understand what you mean by centered box.

For every coordinate in resized tensor we first find the applicable point\location in the original tensor and then take 2 coordinates left of this and 2 coordinates from right . Now if any of these coordinates fall outside of the input length for that dim then the coefficients for these coordinates is set to 0 in case of exclude outside